### DISCUSSION TITLE
Общие переменные для базовых стилей в ui kit

### DISCUSSION THREAD

**Discussion by KirillSolovyev** [chocofamilyme/choco-ui#20] [Ideas]
## 👋 Привет!
  Во время реализации компонента столкнулся с такой проблемой, что общие стили, которые часто переиспользуются, приходится каждый раз хардкодить. К примеру, закругление у элементов (border-radius: 12px). 

### :x:  Почему это плохо:
- Даже при небольшом изменении общих стилей в дизайне придется их менять во всех компонентах, т.к там хардкод
- При реализации разработчик может совершить ошибку в стилях и элемент будет отличаться визуально
- Нет настроенных стилей и при реализации разработчику приходится смотреть в UiKit стили, что отнимает время

### ➡️ Варианты решения:
- Добавить все эти стили в Css variables
- Добавить TailwindCss и настроить 1 раз конфиг для всего UiKit

#### Мне больше нравится вариант с TailwindCss, т.к 
- В нем намного удобнее настраивать конфиги для UiKit
- Быстрее разработка, т.к не нужно долго думать над названиями  и написанием кастомных классов
- Меньший по размеру Css файл, что делает загрузку стилей быстрее

#### Из минусов: 
- Дополнительная зависимость: разработчикам UiKit нужно будет познакомиться с еще одной технологией
- **Возможно**, немного увеличится время билда

### :question: Предлагаю добавить TailwindCss в проект, т.к плюсы значительно перевешивают минусы и сразу решают несколько проблем, с которыми мы столкнулись

**PS:** Лучше это сделать как можно скорее

@Lootjs

**Comment by Lootjs** [comment]
я все таки склоняюсь к переменным. там тоже не очень большой вклад в размер css файла 🤔

**Comment by Lootjs** [comment]
Можешь чекнуть #31 ?
